pkg/didcomm/common/service/event.go

```diff
@@ -12,30 +12,49 @@ type StateMsgType int
 const (
 	// PreState pre state
 	PreState StateMsgType = iota
+
 	// PostState post state
 	PostState
 )
 
-// StateMsg msg
+// StateMsg is used in MsgEvent to pass the state details to the consumer. Refer service.Event.RegisterMsgEvent
+// for more details.
 type StateMsg struct {
-	Type    StateMsgType
+	// type of the message (pre or post), refer service.StateMsgType
+	Type StateMsgType
+
+	// current state. Refer protocol RFC for possible states.
 	StateID string
-	Msg     DIDCommMsg
-	// Properties contains different values from different protocols
-	Properties map[string]interface{}
+
+	// DIDComm message along with message type
+	Msg DIDCommMsg
+
+	// Properties contains value based on specific protocol. The consumers need to cast this to specific interface
```

Comment:
How do consumers currently segregate these state messages per protocol? Closest I can see is by inspecting the type on the enclosed `DIDCommMsg`. Consider including an additional identifier whose contents are an exact match against the protocol identifiers (eg. against `didexchange.DIDExchange`, `introduce.Introduce`, and so on)

Reply:
This is a very valid point. One channel can be registered against multiple protocols and consumers should be able to distinguish protocol at higher level variable than going inside Msg.

I'll add a id in the event message.

Reply:
[Added](https://github.com/hyperledger/aries-framework-go/compare/cc46c9e091f6091fb11e5c1467352ad67d28c0ea..b836b948f8e18a1341aad88b4edfdb0c0443fb4f#diff-f2de91d20207599740587fa5c3ee13c3R27) a new protocol identifier in event messages.